obsreport/obsreport.go

```diff
@@ -32,36 +32,20 @@ const (
 )
 
 var (
+	// InstanceIDKey is used to identify the process instance of the Collector.
+	InstanceIDKey = "instance_id"
```

Comment:
Nice try :) why do you not use the resource in the oc library that produces the metrics?

Reply:
I wasn't aware that it could be used to provide such functionality... any pointers/examples?

Reply:
There are few things:
1. Service instance id in the resource https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-resource-semantic-conventions.md#service
2. If run in Kubernets we can use the pod.id in the resource.

Not trying to block but this, but want to learn why that is not enough.

Reply:
no worries, better get it right. Regarding:
> If run in Kubernetes we can use the pod.id in the resource.

The issue is that a generic mechanism is desired, ie.: something that can be used to identify the instance independent of the deployment environment. The goal is to have a metric label expressing the process uniqueness such metric dashboards can be built in a generic fashion (ie.: working for various different types of deployment environments.

I will take a look at the OTel resources to see how it can be used for the purpose above.